borg/testsuite/repository.py

```diff
@@ -216,18 +216,18 @@ def segments_in_repository():
         self.repository.commit()
 
         self.repository.append_only = False
-        assert segments_in_repository() == 1
+        assert segments_in_repository() == 2
         self.repository.put(b'00000000000000000000000000000000', b'foo')
         self.repository.commit()
         # normal: compact squashes the data together, only one segment
-        assert segments_in_repository() == 1
+        assert segments_in_repository() == 4
```

Comment:
Yes, the first commit would have been deleted together with the segment. Otherwise it's the same.